src/v2/Components/v2/ArtworkFilter/ArtworkFilters/PriceRangeFilter.tsx

```diff
@@ -1,25 +1,89 @@
-import React from "react"
-import { Flex, Radio, RadioGroup, Toggle } from "@artsy/palette"
+import React, { useState } from "react"
+import {
+  Button,
+  Clickable,
+  Flex,
+  Input,
+  Radio,
+  RadioGroup,
+  Spacer,
+  Text,
+  Toggle,
+} from "@artsy/palette"
 import { useArtworkFilterContext } from "../ArtworkFilterContext"
 import { OptionText } from "./OptionText"
 
+const PRICE_RANGES = [
+  { name: "$50K+", value: "50000-*" },
+  { name: "$25K – $50K", value: "25000-50000" },
+  { name: "$10K – $25K", value: "10000-25000" },
+  { name: "$5K – $10K", value: "5000-10000" },
+  { name: "$1K – $5K", value: "1000-5000" },
+  { name: "$0 – $1,000", value: "0-1000" },
+]
+
+type CustomRange = (number | "*")[]
+
+const DEFAULT_CUSTOM_RANGE: CustomRange = ["*", "*"]
+
+const parseRange = (range?: string) => {
+  return range?.split("-").map(s => {
+    if (s === "*") return s
+    return parseInt(s, 10)
+  })
+}
+
 export const PriceRangeFilter: React.FC = () => {
-  const filterContext = useArtworkFilterContext()
-  const initialRange = filterContext.currentlySelectedFilters().priceRange
+  const { currentlySelectedFilters, setFilter } = useArtworkFilterContext()
+  const { priceRange: initialRange, atAuction } = currentlySelectedFilters()
+
+  const numericInitialRange = parseRange(initialRange)
+
+  const [showCustom, setShowCustom] = useState(false)
+  const [customRange, setCustomRange] = useState<CustomRange>(
+    numericInitialRange ?? DEFAULT_CUSTOM_RANGE
+  )
+
+  const handleClick = () => {
+    setFilter("priceRange", customRange.join("-"))
+  }
+
+  const handleChange = (index: number) => ({
+    currentTarget: { value },
+  }: React.FormEvent<HTMLInputElement>) => {
+    const isOpenEnded = value === "" || value === "0"
+    const isMin = index === 0
+    const isMax = index === 1
+
+    setCustomRange(prevCustomRange => {
+      if (isOpenEnded && isMin) {
+        prevCustomRange[index] = 0
+      } else if (isOpenEnded && isMax) {
+        prevCustomRange[index] = "*"
+      } else {
+        prevCustomRange[index] = parseInt(value, 10)
+      }
+
+      return [...prevCustomRange]
```

Comment:
You have to return a new array rather than the mutated array for this to actually update; which is kind of confusing.